pkgs/applications/networking/instant-messengers/element/element-desktop.nix

```diff
@@ -104,7 +110,11 @@ mkYarnPackage rec {
     '';
   };
 
-  passthru.updateScript = ./update.sh;
+  passthru = {
+    updateScript = ./update.sh;
+    # keytar is optional while seshat isn't
+    inherit keytar;
```

Comment:
That does not explain why we need to passthru it and not the other.

Reply:
You wrote that nix ignores attributes on derivations if they aren't directly references in other attached environment variables. Can you point to documentation which describes how that is done and (assuming that assumption is correct) why `passthru` is ever necessary?

Reply:
ok, I have checked the hypothesis that this might be unnecessary (https://github.com/zseri/nixpkgs/commit/7ec54080f110120741bd47f69d4231d8693e4a94), and nix still tries to build keytar, which fails if libsecret isn't available, so this workaround is indeed necessary.